debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java

```diff
@@ -1008,6 +1008,51 @@ public static OversizeHandlingMode parse(String value, String defaultValue) {
                     "for data change, schema change, transaction, heartbeat event etc.")
             .withDefault(DefaultTopicNamingStrategy.class.getName());
 
+    public static final Field MONGODB_MULTI_TASK_ENABLED = Field.create("mongodb.multi.task.enabled")
+            .withDisplayName("Enable Mongo DB multi-task")
+            .withType(Type.BOOLEAN)
+            .withWidth(Width.SHORT)
+            .withImportance(Importance.MEDIUM)
+            .withDefault(false)
+            .withValidation(Field::isBoolean)
+            .withDescription("Should the multi-task connector setup be enabled to the MongoDB instance, only valid if the connector manages a single replset.");
+
+    public static final Field MONGODB_MULTI_TASK_GEN = Field.create("mongodb.multi.task.gen")
+            .withDisplayName("Mongo DB multi-task generation")
+            .withType(Type.INT)
+            .withWidth(Width.SHORT)
+            .withImportance(Importance.MEDIUM)
+            .withDefault(-1)
+            .withValidation(Field::isInteger)
+            .withDescription("Define multi-task generation to be enabled on MongoDB instance");
+
+    public static final Field MONGODB_MULTI_TASK_PREV_TASKS = Field.create("mongodb.multi.prev.tasks.max")
```

Comment:
Are the `prev` and even `gen` options really necessary? Can't we store the number of tasksin the offsets and detect the change based on comparison of value from offests with the one configured in connector config?

Reply:
`prev_gen` is here because we might not want to start at `generation - 1`. Suppose there was some problem with the current generation, we might want to start a new generation starting at the previous generation. Ex: generation 0 is fine, generation 1 is problematic, so we start generation 2 from generation 0's offsets. For this to work `gen` and `prev_gen` need to be explicitly defined.

Moving `gen` into the offset creates an issue in the above scenario because if we don't include the generation in the partition then we won't be able to get the previous generation's offsets.